### PR TITLE
Add FindAllSpecTypes and FindAllStatusTypes

### PR DESCRIPTION
**Pull request by matthchr**
**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

**Comment by matthchr** [comment]
More prep work for secrets

File: v2/tools/generator/internal/astmodel/types.go [hunk 1]
```diff
@@ -457,3 +457,42 @@ func FindStatusTypes(types Types) Types {
...
+		}
+	}
+
+	return result, nil
+}
+
+// FindAllSpecTypes finds all spec types and all types referenced by those spec types.
+// This differs from FindSpecTypes in that it finds not only the top level spec types but
+// also the types which the top level types are built out of.
+func FindAllSpecTypes(types Types) (Types, error) {
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astmodel/types.go:487]: The naming here feels misleading - it doesn't "Find all spec types" it finds "All types that are specs and all the types they reference too".

File: v2/tools/generator/internal/astmodel/types.go [hunk 2]
```diff
@@ -457,3 +457,42 @@ func FindStatusTypes(types Types) Types {
...
+	specTypes := FindSpecTypes(types)
+	return FindConnectedTypes(types, specTypes)
+}
+
+// FindAllStatusTypes finds all status types and all types referenced by those spec types.
+// This differs from FindStatusTypes in that it finds not only the top level spec types but
+// also the types which the top level types are built out of.
+func FindAllStatusTypes(types Types) (Types, error) {
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/astmodel/types.go:495]: Same concern here about naming.

File: v2/tools/generator/internal/astmodel/types.go (hunk 1 above)
Comment (matthchr) [reply v2/tools/generator/internal/astmodel/types.go:487]: Hmm yep - that's a good point. 

Some ideas...:
1. `FindSpecDependentTypes`
2. `FindReachableSpecTypes`
3. `FindConnectedSpecTypes` (or `FindSpecConnectedTypes`)
4. `FindSpecTypeGraph`

I think I like `FindSpecConnectedTypes`.

Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/types.go:487]: `FindSpecConnectedTypes()` is pretty good.

Comment (matthchr) [reply v2/tools/generator/internal/astmodel/types.go:487]: Fixed this

File: v2/tools/generator/internal/astmodel/types.go (hunk 2 above)
Comment (matthchr) [reply v2/tools/generator/internal/astmodel/types.go:495]: Fixed

**Review by theunrepentantgeek — APPROVED** [review]

**Pull request merged**